src/Plugin/Replace/Payload.php

```diff
@@ -80,30 +78,26 @@ public static function fromRequest(Request $request): static {
 	 * @return bool
 	 */
 	public static function hasMatch(Request $request): bool {
-
-		if ($request->format->value === RequestFormat::SQL->value) {
-			$payload = static::$sqlQueryParser::parse(
-				$request->payload,
-				fn($request) => (
-					str_contains($request->error, "P01: syntax error, unexpected SET, expecting VALUES near '")
-					&& stripos($request->payload, 'replace') !== false
-					&& stripos($request->payload, 'set') !== false
-					&& stripos($request->payload, 'where') !== false),
-				$request
-			);
-
-			if (isset($payload['REPLACE'])
-				&& isset($payload['SET'])
-				&& isset($payload['WHERE'][0]['base_expr']) && $payload['WHERE'][0]['base_expr'] === 'id'
-				&& isset($payload['WHERE'][1]['base_expr']) && $payload['WHERE'][1]['base_expr'] === '='
-				&& isset($payload['WHERE'][2]['base_expr']) && is_numeric($payload['WHERE'][2]['base_expr'])
-				&& sizeof($payload['WHERE']) === 3) {
-				return true;
-			}
-			return false;
+		if ($request->format->value === RequestFormat::JSON->value && str_contains($request->path, '/_update/')) {
+			// We process Elastic-like update requests no matter what
+			return true;
 		}
-
-		return str_contains($request->path, '/_update/');
+		$payload = static::$sqlQueryParser::parse(
+			$request->payload,
+			fn($request) => (
+				str_contains($request->error, "P01: syntax error, unexpected SET, expecting VALUES near '")
+				&& stripos($request->payload, 'replace') !== false
```

Comment:
I think it's a good idea to double-check that the query starts with "replace." Validating with an expression that checks if the index is equal to 0 would be preferred.